codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttProperties.java

```diff
@@ -190,20 +201,29 @@ public BinaryProperty(int propertyId, byte[] value) {
     }
 
     public MqttProperties() {
-        this(new HashMap<Integer, MqttProperty>());
+        this(true);
     }
 
-    private MqttProperties(Map<Integer, MqttProperty> props) {
-        this.props = props;
+    private MqttProperties(boolean canModify) {
+        this.canModify = canModify;
     }
 
-    private final Map<Integer, MqttProperty> props;
+    private IntObjectHashMap<MqttProperty> props;
```

Comment:
Why not make it final and avoid further null checks?

Reply:
because IntObjectHashMap is different from HashMap: the former allocate upfront arrays internally, while the latter, nope